### PR TITLE
feat(kuma-cp): shadow mode for policies

### PR DESCRIPTION
**Pull request by lobkovilya**
Implementation of the [madr-40](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/040-transition-to-new-policies.md)

Does **not** cover:
* Global CP support (`_config` endpoint works only on Zone CP)
* no support for `_rules` + `shadow`
* **upd:** no support for `kumactl` in this PR, I'm going to open another one

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/api-server/resource_endpoints.go [hunk 1]
```diff
@@ -108,12 +115,17 @@ func (r *resourceEndpoints) addFindEndpoint(ws *restful.WebService, pathPrefix s
...
 			Returns(404, "Not found", nil))
 	}
-	if r.descriptor.Name == mesh.DataplaneType || r.descriptor.Name == mesh.MeshGatewayType {
+	if r.descriptor.Name == core_mesh.DataplaneType || r.descriptor.Name == core_mesh.MeshGatewayType {
 		ws.Route(ws.GET(pathPrefix+"/{name}/_rules").To(r.rulesForResource()).
 			Doc(fmt.Sprintf("Get matching rules %s", r.descriptor.Name)).
 			Param(ws.PathParameter("name", fmt.Sprintf("Name of a %s", r.descriptor.Name)).DataType("string")).
 			Returns(200, "OK", nil).
 			Returns(404, "Not found", nil))
+		ws.Route(ws.GET(pathPrefix+"/{name}/_config").To(r.configForProxy()).
```
Comment (lobkovilya) [inline pkg/api-server/resource_endpoints.go:124]: I have to add a stub for Global CP as it's not supported at this moment.

File: pkg/api-server/resource_endpoints.go [hunk 2]
```diff
@@ -646,6 +658,92 @@ func (r *resourceEndpoints) matchingDataplanesForPolicy() restful.RouteFunction
...
+	}
+}
+
+func (r *resourceEndpoints) configForProxyParams(request *restful.Request) (*api_types.GetMeshesMeshDataplanesNameConfigParams, error) {
+	params := &api_types.GetMeshesMeshDataplanesNameConfigParams{
+		Shadow:  pointer.To(false),
+		Include: &[]api_types.GetMeshesMeshDataplanesNameConfigParamsInclude{},
+	}
+
+	if shadow := request.QueryParameter("shadow"); shadow != "" {
```
Comment (lukidzi) [inline pkg/api-server/resource_endpoints.go:725]: Do we need to support it in `kumactl` or it works out of the box?

Comment (lobkovilya) [reply pkg/api-server/resource_endpoints.go:725]: We need support in `kumactl`, I'm going to add this in the next PR

File: pkg/core/xds/inspect/config.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	proxy, err := proxyBuilder.Build(ctx, model.ResourceKey{Name: name, Mesh: p.mesh()}, p.meshContext)
+	if err != nil {
+		return nil, err
+	}
+
+	envoyCtx := xds_context.Context{
+		Mesh: p.meshContext,
+		ControlPlane: &xds_context.ControlPlaneContext{
+			CLACache: &cla.Retriever{},
+			Secrets:  &xds.TestSecrets{},
```
Comment (lukidzi) [inline pkg/core/xds/inspect/config.go:59]: I think we shouldn't use in the code `TestSecret` as it is a test resource `"github.com/kumahq/kuma/pkg/test/xds"`

Comment (lobkovilya) [reply pkg/core/xds/inspect/config.go:59]: Ah right, I'll create NoopSecrets

**Review by lukidzi — APPROVED** [review]

**Pull request merged**